src/test/kotlin/com/github/imflog/schema/registry/compatibility/CompatibilityTaskTest.kt

```diff
@@ -115,6 +121,8 @@ class CompatibilityTaskTest {
 
             schemaRegistry {
                 url = 'http://localhost:$REGISTRY_FAKE_PORT/'
+                credentials.username = '$username'
```

Comment:
Can you add this in a separate test, like this we still validate that it still works without auth.

Reply:
`RegistryClientWrapper` is creating a singleton of SchemaRegistryClient, when executing all tests SchemaRegistryClient is initialized only once by the first executing test and the SchemaRegistryClient will be using whatever was the credential setting for the first test and same is applied for all other tests, so executing test with and without auth is not feasible unless I makes changes to `RegistryClientWrapper` to reinitialize SchemaRegistryClient if there are any updates to url or auth. What do you think?

Reply:
You are right about the fact that the registry client is a singleton but in the test context, we use this in the test itself :
```
val result: BuildResult? = GradleRunner.create()
                .withGradleVersion("4.9")
                .withProjectDir(folderRule.root)
                .withArguments(TEST_SCHEMAS_TASK)
                .withPluginClasspath()
                .withDebug(true)
                .build()
```
So the registry client is built for each test I think.

Reply:
I ran all test with debug, the `registryClient` variable in RegistryClientWrapper class is `null` for the first executing test. I made a change to re-initialize registryClient if the input config have changed